fenix/app/src/main/java/org/mozilla/fenix/AppTabMiddleware.kt

```diff
@@ -0,0 +1,119 @@
+package org.mozilla.fenix
+
+import mozilla.components.browser.state.action.ContentAction
+import mozilla.components.browser.state.state.SessionState
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.base.profiler.Profiler
+import mozilla.components.feature.search.SearchUseCases
+import mozilla.components.feature.session.SessionUseCases
+import mozilla.components.feature.tabs.TabsUseCases
+import mozilla.components.lib.state.Middleware
+import mozilla.components.lib.state.MiddlewareContext
+import mozilla.components.support.ktx.kotlin.isUrl
+import mozilla.components.support.ktx.kotlin.toNormalizedUrl
+import org.mozilla.fenix.browser.browsingmode.BrowsingMode
+import org.mozilla.fenix.browser.browsingmode.BrowsingModeManager
+import org.mozilla.fenix.components.appstate.AppAction
+import org.mozilla.fenix.components.appstate.AppState
+import org.mozilla.fenix.components.appstate.Screen
+import org.mozilla.fenix.utils.Settings
+
+class AppTabMiddleware(
+    private val browserStore: BrowserStore,
+    private val settings: Settings,
+    private val tabsUseCases: TabsUseCases,
+    private val sessionUseCases: SessionUseCases,
+    private val searchUseCases: SearchUseCases,
+    private val profiler: Profiler? = null,
+) : Middleware<AppState, AppAction> {
+    override fun invoke(
+        context: MiddlewareContext<AppState, AppAction>,
+        next: (AppAction) -> Unit,
+        action: AppAction
+    ) {
+        next(action)
+        when (action) {
+            is AppAction.ChangeScreen -> {
+                if (action.screen is Screen.Browser) loadBrowserTab(action.screen)
+            }
+            else -> Unit
+        }
+    }
+
+    private fun loadBrowserTab(screen: Screen.Browser) {
+        val startTime = profiler?.getProfilerTime()
+        val mode = BrowsingMode.Normal
+//        val mode = browsingModeManager.mode
+
+
+//        val private = when (mode) {
+//            BrowsingMode.Private -> true
+//            BrowsingMode.Normal -> false
+//        }
```

Comment:
This isn't available because the `BrowsingModeManager` is currently owned by the `HomeActivity` instead of `Components`, where the middleware is initialized.

I already have an in-progress patch to remove `BrowsingModeManager` (see [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1861459)) and move this state into the `AppStore` which would allow this middleware to read the state directly.